### DISCUSSION TITLE
Speeding up builds by keeping results

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD-flow-scripts#1339] [General] [answered]
There are some results that it might be useful to keep in source control. The motivation can be to save execution time, but also to make an explicit choice about when things should change.

Some examples:

- synthesis result
- macro layout
- pin layout

My thinking is that this could be supported by adding some KEEP_* environment variables. If these are set, then the rule in ORFS Makefile would be changed to copy the file from the config.mk folder if it exists and if it doesn't exist, generate it, then copy it.

The user could then keep this result around by adding the results to source control.

The "result" should be source control friendly: a .tcl file for manual layout for macro layout, pin layout. Synthesis result is a large .v file, which git compresses.

Another motivation is that e.g. syntehsis is run when the .sdc file changes, but if one is working on CTS and syntehsis and macro layout is slow, then it's nice to cache those stages, even if it one gets a less accurate result when iterating on CTS.

**Comment by vvbandeira** [comment, marked as answer]
FYI, we already have the option to cache the netlist for synthesis. You need to set the CACHED_NETLIST variable.
Reference:
https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/blob/master/flow/scripts/synth_preamble.tcl#L3-L9

**Comment by vvbandeira** [comment]
@oharboe
Are you satisfied with caching only the netlist? Would you still be interested in caching other results/stage files?

**Reply by oharboe** [reply]
Yes, but I want to tinker with the current implementation to propose a different workflow. This is not super high on my list, so I will see when/if I get around to it.
